Patch origin: in-house
Patch status: Solaris-specific; not suitable for upstream

Customize conf file for Solaris.

Updated 2025-05-26 for Apache2 modular configuration

--- modsecurity-2.9.7/modsecurity-24.conf-recommended.~1~	2025-05-26 12:01:49.165602546 +0000
+++ modsecurity-2.9.7/modsecurity-24.conf-recommended	2025-05-26 12:18:37.021045045 +0000
@@ -1,3 +1,5 @@
+<IfModule mod_security2.c>
+
 # -- Rule engine initialization ----------------------------------------------
 
 # Enable ModSecurity, attaching it to every transaction. Use detection
@@ -147,13 +149,13 @@
 # This default setting is chosen due to all systems have /tmp available however, 
 # this is less than ideal. It is recommended that you specify a location that's private.
 #
-SecTmpDir /tmp/
+SecTmpDir /var/apache2/2.4/modsec/tmp/
 
 # The location where ModSecurity will keep its persistent data.  This default setting 
 # is chosen due to all systems have /tmp available however, it
 # too should be updated to a place that other users can't access.
 #
-SecDataDir /tmp/
+SecDataDir /var/apache2/2.4/modsec/tmp/
 
 
 # -- File uploads handling configuration -------------------------------------
@@ -162,7 +164,7 @@
 # location must be private to ModSecurity. You don't want other users on
 # the server to access the files, do you?
 #
-#SecUploadDir /opt/modsecurity/var/upload/
+#SecUploadDir /var/apache2/2.4/modsec/upload/
 
 # By default, only keep the files that were determined to be unusual
 # in some way (by an external inspection script). For this to work you
@@ -182,7 +184,7 @@
 # The default debug log configuration is to duplicate the error, warning
 # and notice messages from the error log.
 #
-#SecDebugLog /opt/modsecurity/var/log/debug.log
+#SecDebugLog /var/apache2/2.4/logs/modsec_debug.log
 #SecDebugLogLevel 3
 
 
@@ -202,10 +204,10 @@
 # assumes that you will use the audit log only ocassionally.
 #
 SecAuditLogType Serial
-SecAuditLog /var/log/modsec_audit.log
+SecAuditLog /var/apache2/2.4/logs/modsec_audit.log
 
 # Specify the path for concurrent audit logging.
-#SecAuditLogStorageDir /opt/modsecurity/var/audit/
+#SecAuditLogStorageDir /var/apache2/2.4/modsec/audit/
 
 
 # -- Miscellaneous -----------------------------------------------------------
@@ -227,7 +229,7 @@
 # to properly map encoded data to your language. Properly setting
 # these directives helps to reduce false positives and negatives.
 #
-SecUnicodeMapFile unicode.mapping 20127
+SecUnicodeMapFile /var/apache2/2.4/modsec/unicode.mapping 20127
 
 # Improve the quality of ModSecurity by sharing information about your
 # current ModSecurity version and dependencies versions.
@@ -238,3 +240,4 @@
 # setting On, as there is no active receiver for the information.
 SecStatusEngine Off
 
+</IfModule>
